src/context.rs

```diff
@@ -2598,4 +2620,19 @@ mod tests {
             "Internal error: Error initializing WASM + MessagePack UDF \"invalidfn\": Internal(\"Error loading WASM module: failed to parse WebAssembly module"));
         Ok(())
     }
+
+    #[tokio::test]
+    #[ignore]
+    async fn test_drop_function() -> Result<()> {
+        let sf_context = in_memory_context().await;
+
+        // Source: https://gist.github.com/going-digital/02e46c44d89237c07bc99cd440ebfa43
+        let plan = sf_context
+            .plan_query(r#"DROP FUNCTION nonexistentfunction"#)
+            .await;
+        assert!(plan.is_err());
```

Comment:
You could also match on the error message.

(also the comment above is obsolete.)

Reply:
Are you saying to match on the string contents?

```
$ ./delete_udf.sh
HTTP/1.1 400 Bad Request
vary: Authorization, Content-Type, Origin, X-Seafowl-Query
content-length: 230
date: Sat, 05 Aug 2023 07:11:59 GMT

Internal error: Internal SQL error: "no rows returned by a query that expected to return at least one row". This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker%                    
```

e.g. 'no rows returned'? 

I had the impression `plan.is_err()` might be sufficient to match on, but not 100% sure the pros/cons - would be curious your thoughts.

Reply:
The idea is that the error we hit is also the one we expect. The one above is directly from sqlite/postgres so we'd need to translate it to match the specific case being handled here i.e. function didn't exist in the first place.

I think you'd need to match for `Err(RepositoryError::SqlxError(sqlx::error::Error::RowNotFound))` in the catalog `drop_function` call, and then if `if_exists` is true just swallow that error, otherwise surface it with a custom message.

Reply:
Now that we restored `FunctionNotFound` inside catalog.rs, I updated the tests to specifically check for the string literal. Marking resolved, please re-open as needed